airbyte-cdk/bulk/core/base/src/main/kotlin/io/airbyte/cdk/StreamNamePair.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk
+
+import io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair
+import io.airbyte.protocol.models.v0.StreamDescriptor
+
+/**
+ * [StreamNamePair] is equivalent to [AirbyteStreamNameNamespacePair].
+ *
+ * This exists to avoid coupling the Bulk CDK code too closely to the Airbyte Protocol objects.
+ */
+data class StreamNamePair(val name: String, val namespace: String?) {
```

Comment:
We're modeling this already in load land as `StreamDescriptor`. I'll make a note to switch to using yours.

Reply:
I saw your `DestinationStream.Descriptor` definition. I like your `asProtocolObject` method name more, I'll carry that over.

FYI, for entirely historical reasons, namespace comes typically comes after name.

Reply:
>  I like your asProtocolObject method name more, I'll carry that over.

Actually I'm not going to do that, because this maps to both `AirbyteStreamNameNamespacePair` and `StreamDescriptor` protocol objects.

Reply:
not a massive fan of the `StreamNamePair` name. `StreamIdentifier`?

Reply:
I don't think we need to carry over weird argument orders. usually, they would go the other way around

Reply:
> StreamIdentifier

Thanks, I like that, I'll submit a PR.

Reply:
> I don't think we need to carry over weird argument orders. usually, they would go the other way around

Naturally, but this is the way it is in the protocol object methods. I'm worried about bugs caused by flipping over the name and the namespace positions.

Reply:
Do you think it's worth taking that hit? I'm on the fence, personally, so I'm not saying no.

Reply:
Oh, that's a very good point!! Hadn't thought of that. FYI I think you mean catalog and not config but your point still stands. Lemme see if I can fix this.